internal/generate/run.go

```diff
@@ -16,13 +16,71 @@ package generate
 
 import (
 	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/googleapis/generator/internal/gitrepo"
 )
 
 func Run(ctx context.Context, arg ...string) error {
 	cfg := &config{}
-	_, err := parseFlags(cfg, arg)
+	cfg, err := parseFlags(cfg, arg)
+	if err != nil {
+		return err
+	}
+	googleapisRepo, err := cloneGoogleapis(ctx)
+	if err != nil {
+		return err
+	}
+	languageRepo, err := cloneLanguageRepo(ctx, cfg.language)
 	if err != nil {
 		return err
 	}
-	return nil
+	return runDocker(googleapisRepo.Dir, languageRepo.Dir, cfg.api)
+}
+
+const googleapisURL = "https://github.com/googleapis/googleapis"
+
+func cloneGoogleapis(ctx context.Context) (*gitrepo.Repo, error) {
+	repoPath := filepath.Join(os.TempDir(), "/generator-googleapis")
+	return gitrepo.CloneOrOpen(ctx, repoPath, googleapisURL)
+}
+
+func cloneLanguageRepo(ctx context.Context, language string) (*gitrepo.Repo, error) {
+	languageRepoURL := fmt.Sprintf("https://github.com/googleapis/google-cloud-%s", language)
+	repoPath := filepath.Join(os.TempDir(), fmt.Sprintf("/google-cloud-%s", language))
+	return gitrepo.CloneOrOpen(ctx, repoPath, languageRepoURL)
+}
+
+const dotnetContainerDir = "containers/dotnet"
+
+func runDocker(googleapisDir, languageDir, api string) error {
+	if err := runCommand("docker", "build", dotnetContainerDir, "-t", "picard"); err != nil {
+		return err
+	}
+	args := []string{
+		"run",
+		"-v", fmt.Sprintf("%s:/apis", googleapisDir),
+		"-v", fmt.Sprintf("%s:/output", languageDir),
+		dotnetContainerDir,
```

Comment:
This should be "picard" instead of the container directory. (It's the tag of the already-built image.)

Reply:
updated